index.js

```diff
@@ -93,7 +93,7 @@ class PQueue {
 			const run = () => {
 				this._pendingCount++;
 
-				Promise.resolve(fn()).then(
+				Promise.resolve((async () => fn())()).then(
```

Comment:
Async functions aren't supported in the versions of `node` that `p-queue` supports.  Wanna use `p-try`?
```
pTry(fn).then(
```

Reply:
That would definitely fix the problem. I did some [benchmarking](https://pastebin.com/raw/K36MZUK2): using promises/async it's about ~10% slower than try/catch (CPU: i7-7700K).